src/scss/_functions.scss

```diff
@@ -108,37 +108,42 @@
 /// @param {String} $color-name - the name of the color to be shaded
 /// @param {Number} $brightness - the brightness value of the new color, 0-100
 @function oColorsGetTone($color-name, $brightness) {
+	// Find palette colour information.
 	$color: oColorsByName($color-name);
-	$hue: hue($color);
-	// Find a tone from a base of saturation which differs per colour.
-	$saturation: map-get($_o-colors-tone-saturation, $color-name);
-
-	// Validate the given colour allows different tones.
-	$has-tone-config: map-has-key($_o-colors-tone-saturation, $color-name);
-	@if (not $has-tone-config and ($color-name == 'white' or $color-name == 'black')) {
-		@return _oColorsError('"#{$color-name}" does not support tones. ' +
-			'Use a mix instead: ' +
-			'`oColorsMix(\'#{$color-name}\', $percentage: #{$brightness})`');
-	}
-	@if (not $has-tone-config) {
-		@return _oColorsError('"#{$color-name}" does not support tones. ' +
-			'Consider using the `oColorsMix` function to mix with black to ' +
-			'darken or mix with white to lighten.');
-	}
+	$color-map: map-get($_o-colors-palette, $color-name);
+	$color-meta: map-get($color-map, 'meta');
 
 	// Validate brightness.
 	@if(type-of($brightness) != 'number' or $brightness > 100 or $brightness < 0) {
 		@return _oColorsError('"$brightness" must be a number between 0 and 100.');
 	}
 
-	@if $brightness == 0 {
-		@return hsla(0, 0, 0, 1);
-	} @else {
-		$hsl-luminance: ($brightness/2) * (2 - ($saturation/100));
-		$hsl-saturation: ($brightness * $saturation) / if($hsl-luminance < 50, $hsl-luminance * 2, 200 - $hsl-luminance * 2);
+	// Error for white and black colours which have no hue, they cannot be toned.
+	@if (hue($color) == 0) {
```

Comment:
I don't think this line is correct.
Black is any hue + any saturation + 0 brightness
White is any hue + 0 saturation + 100 brightness

Reply:
If it has a hue we can increase saturation and brightness to get a tone!

Reply:
I guess that means the comment above it is wrong then?

Reply:
I think that's better now

Reply:
isnt a hue of 0 with saturation 100% red?

Reply:
Oh you're right, thanks chee! black, white, grey returns a `0deg` hue value but so does red. I also need to check the saturation is `0%`